app/models/products/user_note_mode.rb

```diff
@@ -0,0 +1,49 @@
+module Products
+
+  class UserNoteMode
+
+    attr_reader :raw_value
+
+    def self.values
+      valid_values.map { |value| new(value) }
+    end
+
+    def self.valid_values
+      %w(hidden optional required).freeze
+    end
+
+    def initialize(raw_value)
+      @raw_value = raw_value
+      freeze
```

Comment:
I did do a bit of a double-take that we weren't doing that check here, although it makes sense that we would have the validation on product.